text/0000-remove-windows-containers-support.md

```diff
@@ -0,0 +1,97 @@
+# Meta
+[meta]: #meta
+- Name: Remove Windows Containers Support
+- Start Date: 2024-04-11
+- Author(s): aidan.delaney
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+Retain Windows Containers support in the buildpacks specification, but remove Windows Containers support from `lifecycle`, `pack` CLI, and documentation.
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that buildpack authors or other interested parties may not be familiar with.
+
+* [Windows Containers](https://learn.microsoft.com/en-us/virtualization/windowscontainers/about/): Containers derived from Windows Server Containers.  For example: `mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-ltsc2022`
+
+# Motivation
+[motivation]: #motivation
+
+- There are no known users of Windows Container support within the buildpacks ecosystem.
+- Supporting Windows Containers is straightforward at a specification level.  However, the CNB project has neither the time nor the expertise to continue to support Windows Containers in our tooling.
+- Removing Windows Container support from CNB tooling will speed up development, simplify testing and impact no existing users.
+- Establish platform parity between `pack` and `kpack`.  `kpack` has never supported building Windows Containers.
+
+# What it is
+[what-it-is]: #what-it-is
+
+- We want to remove Windows Containers related documentation, but clearly state that the specification is platform neutral.
```

Comment:
How hard will it be to make a platform neutral specification format, i.e. POSIX vs. windows?

Reply:
We're hoping to maintain the status-quo in the specification where we currently state things like 

* "MUST NOT be run for Windows builds" (https://github.com/buildpacks/spec/blob/main/buildpack.md#purpose-2)
* "...with the OS path list separator (e.g. : on Linux, ; on Windows)." (https://github.com/buildpacks/spec/blob/main/buildpack.md#targets)
* "For Windows-based images..." (https://github.com/buildpacks/spec/blob/main/buildpack.md#targets-1)

It will be difficult to keep in mind the POSIX/Windows differences when changing the specification.  However, I suspect this will be no-more difficult than currently as we have no known Windows containers users.